keps/sig-apps/0032-portable-service-definitions.md

```diff
@@ -0,0 +1,157 @@
+
+The next step is to work out the details on a common service and an initial process by which future services can be added. To work out the details we will start with MySQL and go through the process to make it work as managed by Kubernetes and each of the top three public clouds as defined by adoption. Public clouds and other cloud platforms following the top three are welcome to be involved in the process but are not required in the process.
+
+Before this KEP can move to being implementable at least 2 services need to go through the process of being implemented to prove out the process elements of this system.
+
+### Risks and Mitigations
+
+Two known risks include:
+
+1. The access controls and relationships between accounts and services. How will proper user and tenancy information be passed to clouds that require this form of information?
+2. Details required, when requesting a service, can vary between cloud providers that will implement this as a SaaS. How can that information be commonized or otherwise handled?
+
+## Graduation Criteria
+
+The following are the graduation criteria:
+
+- 5 organizations have adopted using the portable service definitions
+- Service definitions for at least 3 services have been created and are in use
+- Documented exists explaining how a controller implementer can use the CRDs and secrets to create a controller of their own
+- Service consumer documentation exists explaining how to use portable service definitions within their applications
+- A documented process for bringing a new service from suggestion to an implementable solution
+
+## Implementation History
+
+- the _Summary_, _Motivation_, _Proposal_ sections developed
+
+## Drawbacks [optional]
+
+Why should this KEP _not_ be implemented.
+
+## Alternatives [optional]
+
+An alternative is to modify the service catalog to leverage CRDs and return common secret credentials. Drawbacks to this are that it would be a complicated re-write to the service catalog, according to the service catalog team, and the solution would still require the open service broker (OSB) to be implemented in all environments (e.g., Minikube) even where simpler models (e.g., a controller) could be used instead. The solution proposed here could work with the service catalog in environments it makes sense and use other models in other environments. The focus here is more on the application operator experience working with services than all of the implementations required to power it.
+
+## Infrastructure Needed [optional]
+
+The following infrastructure elements are needed:
+
+- A new subproject under SIG Apps for organizational purposes
+- A git repository, in the `kubernetes-sigs` organization, to host the CRD and Secrets schemas along with the Kubernetes provided controller
+- Testing infrastructure to continuously test the codebase
```

Comment:
Is a slack channel needed?

Reply:
Can we just use the sig apps slack channel? Is there a reason for a new one yet?

Reply:
Nope, was just curious!